File: python/jail-roster.py
#!/usr/bin/env python

# Written for Python 2.x
# Tested on
# Windows 7 x64 - Python 2.7.6

import urllib
import json

if __name__ == "__main__":
  # get the arrest data
  roster = urllib.urlopen("http://api.canyonco.org/Sheriff-Beta/JailRosterByLetter/A").read()
  # parse the JSON
  jsonData = json.loads(roster)

  # If the first element's Error is not empty it will be the error message
  # not exactly restful, but it was easy
  error = jsonData[0]["Error"]

  if error == '':
    # loop through the inmates
    for inmate in jsonData:
      print "%s, %s" % (inmate["LastName"].title(), inmate["FirstName"].title())

      # loop through the arrests
      for arrest in inmate["Arrests"]:
        print "  Arrest Date: %s" % arrest["ArrestDate"]

      # loop through the charges
      for charge in inmate["Charges"]:
        print "  Charge: %s: %s" % (charge["StatuteCode"], charge["StatuteDesc"])

  else:
    print "Error getting data", error

File: ruby/jail-roster.rb
#!/usr/bin/env ruby

# Written for Ruby 1.9.x
# Tested on
# Windows 7 x64 - Ruby 1.9.3
# Ubuntu 14.04 LTS (trusty) - Ruby 1.9.3

require 'open-uri'
require 'json'

# get the and parse the arrest data
jsonData = JSON.parse(open("http://api.canyonco.org/Sheriff/JailRosterByLetter/A").read)

# If the first element's Error is not empty it will be the error message
# not exactly restful, but it was easy
if jsonData[0]["Error"].empty?
  
  # loop through the inmates
  jsonData.each do |inmate|
    puts "#{inmate["LastName"].capitalize}, #{inmate["FirstName"].capitalize}"

    # loop through the arrests
    inmate["Arrests"].each do |arrest|
      puts "  Arrest Date: #{arrest["ArrestDate"]}"
    end
    
    # loop through the charges
    inmate["Charges"].each do |charge|
      puts "  Charge: #{charge["StatuteCode"]}: #{charge["StatuteDesc"]}"
    end
  end
else
  puts "Error: " + jsonData[0]["Error"]
end